helix-core/src/chars.rs

```diff
@@ -6,7 +6,7 @@ use crate::LineEnding;
 pub enum CharCategory {
     Whitespace,
     Eol,
-    Word,
+    Word(u8),
```

Comment:
Rather than an integer let's make a new enum for this, "WordCategory" or similar with variants from the comments in `char_is_word`: `Ascii`, `Superscript`, `Subscript`, ....